tests/ckeditor.jsx

```diff
@@ -464,67 +444,138 @@ describe( 'CKEditor Component', () => {
 			const editorInstance = new Editor();
 
-			sandbox.stub( Editor, 'create' ).resolves( editorInstance );
-			sandbox.stub( editorInstance, 'destroy' ).resolves();
+			sinon.stub( Editor, 'create' ).resolves( editorInstance );
+			sinon.stub( editorInstance, 'destroy' ).resolves();
 
-			wrapper = mount( <CKEditor editor={ Editor } /> );
+			await new Promise( ( res, rej ) => {
+				wrapper = mount( <CKEditor
+					editor={ Editor }
+					onInit={ res }
+					onError={ rej } /> );
+			} );
 
-			setTimeout( () => {
-				wrapper.unmount();
-				wrapper = null;
+			const component = wrapper.instance();
 
-				expect( editorInstance.destroy.calledOnce ).to.be.true;
+			expect( component.editor ).is.not.null;
 
-				done();
+			wrapper.unmount();
+			wrapper = null;
+
+			// Wait a cycle.
+			await new Promise( res => setTimeout( res ) );
+
+			expect( component.editor ).is.null;
+		} );
+	} );
+
+	describe( 'watchdog', () => {
+		it( 'should be available for the component', async () => {
+			await new Promise( ( res, rej ) => {
+				wrapper = mount( <CKEditor
+					editor={ Editor }
+					onInit={ res }
+					onError={ rej } /> );
 			} );
+
+			expect( wrapper.instance().watchdog ).to.be.an( 'object' );
 		} );
 
-		it( 'should set to "null" the "editor" property inside the component', done => {
-			const editorInstance = new Editor();
+		it( 'should handle errors and restart the editor', async () => {
+			let onInitCallback;
+			let onErrorCallback;
 
-			sandbox.stub( Editor, 'create' ).resolves( editorInstance );
-			sandbox.stub( editorInstance, 'destroy' ).resolves();
+			function onInit() {
+				onInitCallback();
+			}
 
-			wrapper = mount( <CKEditor editor={ Editor } /> );
+			function onError() {
+				onErrorCallback();
+			}
 
-			setTimeout( () => {
-				const component = wrapper.instance();
+			const onInitSpy = sinon.spy( onInit );
+			const onErrorSpy = sinon.spy( onError );
 
-				expect( component.editor ).is.not.null;
+			wrapper = mount( <CKEditor
+				editor={ Editor }
+				onInit={ onInitSpy }
+				onError={ onErrorSpy } /> );
 
-				wrapper.unmount();
-				wrapper = null;
+			await new Promise( ( res, rej ) => {
+				onInitCallback = res;
+				onErrorCallback = rej;
+			} );
 
-				setTimeout( () => {
-					expect( component.editor ).is.null;
+			const watchdog = wrapper.instance().watchdog;
 
-					done();
-				} );
+			await new Promise( ( res, rej ) => {
+				onInitCallback = res;
+				onErrorCallback = rej;
+
+				watchdog._restart();
 			} );
+
+			sinon.assert.calledTwice( onInitSpy );
+			sinon.assert.notCalled( onErrorSpy );
+
+			const editor1 = onInitSpy.firstCall.args[ 0 ];
+			const editor2 = onInitSpy.secondCall.args[ 0 ];
+
+			expect( editor1 ).to.be.instanceOf( Editor );
+			expect( editor2 ).to.be.instanceOf( Editor );
+
+			expect( editor1 ).to.not.equal( editor2 );
 		} );
 	} );
 } );
+
+describe( 'CKEditor.Context', () => {
```

Comment:
There should be also a test that checks whether the component created by using `CKEditor.Context` works since users can also use this way to use the feature.

Reply:
I don't think so. If this is just the same thing as already tested I think that we can just ensure that the correct thing is exported.

Reply:
Ok, I checked that `MainComponent.Children` works.

Reply:
However, it looks bad if you look at the tree:

```html
<CKEditor.Context>
    <CKEditor>
```

The names suggest that `Context` is a child of `CKEditor`. In fact, `CKEditor` is a child of `Context`.

Reply:
I guess we need to modify the API exported by the package.

```js
import { CKEditor, Context } from '@ckeditor/ckeditor5-react';
```

Looks good. We could also export the `CKEditor` component as default so no breaking changes will be introduced.

```js
// The imports below should works as well:
import CKEditor, { Context } from '@ckeditor/ckeditor5-react';
import CKEditor from '@ckeditor/ckeditor5-react';
```

Reply:
I've added the new package entry point.